guides/source/asset_pipeline.md

```diff
@@ -868,218 +904,348 @@ each asset will have a different file name when it is changed. This way you
+  bundle remove sprockets
+  bundle remove sprockets-rails
+  bundle remove sass-rails
+  ```
 
-Possible options for JavaScript compression are `:terser`, `:closure` and
-`:yui`. These require the use of the `terser`, `closure-compiler` or
-`yui-compressor` gems, respectively.
+2. Delete the `config/assets.rb` and `assets/config/manifest.js` files from your
+   project.
 
-Take the `terser` gem, for example.
-This gem wraps [Terser](https://github.com/terser/terser) (written for
-Node.js) in Ruby. It compresses your code by removing white space and comments,
-shortening local variable names, and performing other micro-optimizations such
-as changing `if` and `else` statements to ternary operators where possible.
+3. If you've already upgraded to Rails 8, then Propshaft is already included in
+   your application, otherwise, install it using `bundle add propshaft`.
 
-The following line invokes `terser` for JavaScript compression.
+4. Remove the `config.assets.paths << Rails.root.join('app', 'assets')` line
+   from your `application.rb` file.
 
-```ruby
-config.assets.js_compressor = :terser
-```
+5. Migrate asset helpers by replacing all instances of asset helpers (e.g.,
+`image_url`) with standard URLs because Propshaft utilizes relative paths. For
+example, image_url("logo.png") will become url("/logo.png).
 
-NOTE: You will need an [ExecJS](https://github.com/rails/execjs#readme)
-supported runtime in order to use `terser`. If you are using macOS or
-Windows you have a JavaScript runtime installed in your operating system.
+6. If you're relying on Sprockets for transpiling, you'll need to switch to a
+   Node-based transpiler like Webpack, esbuild, or Vite. You can use the
+   `jsbundling-rails` and `cssbundling-rails` gems to integrate these tools into
+   your Rails application.
 
-NOTE: The JavaScript compression will also work for your JavaScript files when you are loading your assets through `importmap-rails` or `jsbundling-rails` gems.
+For more information, you can read the [detailed guide on how to migrate from
+Sprockets to
+Propshaft](https://github.com/rails/propshaft/blob/main/UPGRADING.md).
 
-### GZipping your assets
+### Fingerprinting
 
-By default, gzipped version of compiled assets will be generated, along with
-the non-gzipped version of assets. Gzipped assets help reduce the transmission
-of data over the wire. You can configure this by setting the `gzip` flag.
+Fingerprinting is a technique that makes the name of a file dependent on its
+content. A digest of the file's content is generated and appended to the
+filename. This ensures that when the file content changes, its digest—and
+consequently its filename—also changes. This mechanism is crucial for caching
+assets effectively, as the browser will always load the updated version of an
+asset when its content changes, thereby improving performance.
 
-```ruby
-config.assets.gzip = false # disable gzipped assets generation
-```
+### Asset Precompilation and Digesting
```

Comment:
Yup! Thanks for calling this out. This and a few other sections were repeated because of my structure, however, I’ve re-ordered the sections so that the flow feels much more natural, and it helped me catch and remove some duplications.